src/Microsoft.OData.Client/Serialization/HttpWebRequestMessage.cs

```diff
@@ -480,122 +495,24 @@ internal void AfterSendingRequest2Event()
 #endif
         }
 
-        /// <summary>
-        /// Create an instance of HttpWebRequest from the given IODataRequestMessage instance. This method
-        /// is also responsible for firing SendingRequest event.
-        /// </summary>
-        /// <param name="method">Http Method.</param>
-        /// <param name="requestUrl">Request Url.</param>
-        /// <param name="args">DataServiceClientRequestMessageArgs instance.</param>
-        /// <returns>an instance of HttpWebRequest.</returns>
-        [SuppressMessage("Microsoft.Usage", "CA1801:ReviewUnusedParameters", MessageId = "args", Justification = "the parameter is used in the SL version.")]
-        private static HttpWebRequest CreateRequest(string method, Uri requestUrl, DataServiceClientRequestMessageArgs args)
+        public void Dispose()
         {
-            HttpWebRequest httpRequest = (HttpWebRequest)WebRequest.Create(requestUrl);
-#if PORTABLELIB
-            // Set same default as Silverlight when running silverlight in .Net Portable
-            if (IsRunningOnSilverlight)
-            {
-                httpRequest.UseDefaultCredentials = args.UseDefaultCredentials;
-            }
-#else
-            // KeepAlive not available
-            httpRequest.KeepAlive = true;
-#endif
-            httpRequest.Method = method;
-            return httpRequest;
+            this.Dispose(true);

```

Comment:
Can `_messageStream` (and any other disposable object) be disposed in this method or do we leave GC to do it for us?